### PR TITLE
Implement AB testing

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
This PR implements [this issue](https://github.com/Cap-go/capacitor-updater/issues/37)

Here is a recorded demo (Sorry for the length)

https://github.com/Cap-go/capgo/assets/50914789/9ce1ba9e-819b-4ed8-be9d-abeb670480f4

### CODE REVIEW

**Comment by WcaleNieWolny** [comment]
image

Is this okay? I am aware that weak crypto = bad but it does not really matter here since the user can just spam update request until he gets the version he wants

**Comment by WcaleNieWolny** [comment]
Do not merge this just yet please, debounce is broken

**Comment by WcaleNieWolny** [comment]
@riderx Fixed the debounce issue, ready to be merged

**Comment by riderx** [comment]
Can you revert change in CHANGELOG.md android/app/build.gradle ios/App/App.xcodeproj/project.pbxproj and package.json
this shouldn't be changed i think.
After i can merge it

**Comment by WcaleNieWolny** [comment]
@riderx done

**Comment by WcaleNieWolny** [comment]
@riderx Fixed merge conflicts (again, as you added some commits after I did that yesterday)

You can squash commits when you merge

File: supabase/migrations/20230815171919_base.sql [hunk 1]
```diff
@@ -1051,6 +1051,9 @@ CREATE TABLE "public"."channels" (
     "updated_at" timestamp with time zone DEFAULT "now"() NOT NULL,
     "public" boolean DEFAULT false NOT NULL,
     "disableAutoUpdateUnderNative" boolean DEFAULT true NOT NULL,
+    "enableAbTesting" boolean not null default false,
+    "secondaryVersionPercentage" double precision not null default '0'::double precision,
+    "secondVersion" bigint not null default '1883'::bigint,
```
Comment (riderx) [inline supabase/migrations/20230815171919_base.sql:1056]: the default var here seems wrong no ?

Comment (WcaleNieWolny) [reply supabase/migrations/20230815171919_base.sql:1056]: No, it does not. It is the default unknown version. Please see seed.sql. A version with this exact id is created there

Comment (WcaleNieWolny) [reply supabase/migrations/20230815171919_base.sql:1056]: Here is the line I am talking about

https://github.com/WcaleNieWolny/capgo/blob/069db6e75b3e41dd30bf2621c43d235800efdb32/supabase/seed.sql#L46

**Pull request merged**